### PR TITLE
feat: migrate admin ee server to typescript

### PR DESCRIPTION
**Pull request by Marc-Roig**
### What does it do?
Migrate admin ee server to typescript

Practically everything is typed as any, this work added the necessary changes to migrate every file to typescript and for the code to be runable.

For some reason, the project does run on EE but I can not see the EE features on the frontend side. @jhoward1994 I might need your help to check why.

Edit: NVM, I found the issue, we had a commented line on the /admin/init controller

### CODE REVIEW

**Comment by jhoward1994** [comment]
Is it worth making the constants at `packages/core/admin/ee/server/src/config/admin-actions.js` TS as well?

**Comment by Marc-Roig** [comment]
indeed, those should be constants too. good catch.

File: packages/core/admin/ee/server/src/utils/sso-lock.ts [hunk 1]
```diff
@@ -1,10 +1,8 @@
-'use strict';
+import EE from '@strapi/strapi/dist/utils/ee';
+import { isEmpty } from 'lodash/fp';
 
-const { features } = require('@strapi/strapi/dist/utils/ee').default;
-const { isEmpty } = require('lodash/fp');
-
-const isSsoLocked = async (user) => {
-  if (!features.isEnabled('sso')) {
+export const isSsoLocked = async (user: any) => {
```
Comment (jhoward1994) [inline packages/core/admin/ee/server/src/utils/sso-lock.ts:4]: In these utils are you deliberately exporting as a default object and as named exports?

File: packages/core/admin/ee/server/src/utils/persisted-tables.ts [hunk 2]
```diff
@@ -82,7 +84,7 @@ async function getPersistedTables({ strapi }) {
  * @param {Array<string|{ table: string; dependsOn?: Array<{ table: string;}> }>} tableNames
  * @returns {Promise<void>}
  */
-async function setPersistedTables({ strapi }, tableNames) {
+async function setPersistedTables({ strapi }: { strapi: LoadedStrapi }, tableNames: string[]) {
```
Comment (jhoward1994) [inline packages/core/admin/ee/server/src/utils/persisted-tables.ts:87]: string[] doesn't match what we have in the old JSDOC, should we define this as a type?

Comment (Marc-Roig) [reply packages/core/admin/ee/server/src/utils/persisted-tables.ts:87]: true, will update it!

File: packages/core/admin/ee/server/src/utils/sso-lock.ts (hunk 1 above)
Comment (Marc-Roig) [reply packages/core/admin/ee/server/src/utils/sso-lock.ts:4]: yes, I wonder if it's something we should avoid. But some files do import it as a default object and others as named exports, so I am following this pattern, specially for utils and validations files.

**Review by jhoward1994 — APPROVED** [review]

**Pull request merged**